mmaction/models/backbones/tanet.py

```diff
@@ -0,0 +1,113 @@
+import torch.nn as nn
+from torch.utils import checkpoint as cp
+
+from ..common import TAM
+from ..registry import BACKBONES
+from .resnet import Bottleneck, ResNet
+
+
+class TABlock(nn.Module):
+    """Temporal Adaptive Block (TA-Block) for TANet.
+
+    "The temporal adaptive module (TAM) is embedded into ResNet-Block
+    after the first Conv2D, which turns the vanilla ResNet-Block
+    into TA-Block."
+
+    Args:
+        block (nn.Module): Residual blocks to be substituted.
+        num_segments (int): Number of frame segments.
+        tam_cfg (dict | None): Config for temporal adaptive module (TAM).
+            If set to None, it uses ```dict()``` to constrcut TAM.
+            Default: None.
+    """
+
+    def __init__(self, block, num_segments, tam_cfg=None):
+        super().__init__()
+        if tam_cfg is None:
+            tam_cfg = dict()
```

Comment:
Actually, the mutable value can be set as default values, as long as you use `copy.deepcopy()` to protect it. The method is also approved by @innerlee